### PR TITLE
Move to graph studio

### PR DESCRIPTION
**Pull request by tukantje**
This PR does the following;

* Move us to decentralised graph studio from graph hosted service
* Remove Rinkeby
* Update configs to match the new system
* Add quality of life commands to simplify deployment

### CODE REVIEW

File: src/scripts/deploy.js [hunk 1]
```diff
@@ -1,56 +1,90 @@
...
 
-console.log(`Prepare subgraph config for network ${NETWORK}`)
-console.log(`  Generate subgraph.yaml`)
+const AUTH_TOKEN = process.env.AUTH_TOKEN;
+assert(AUTH_TOKEN, "AUTH_TOKEN env var is required");
 
-series([
-  (callback) => exec(`mustache config/${NETWORK}.json subgraph.yaml.mustache > subgraph.yaml`, null, callback),
+const VERSION = process.env.VERSION;
+assert(VERSION, "VERSION env var is required");
```
Comment (anxolin) [inline src/scripts/deploy.js]: If we add envs, shouldb't we document in the README?

Comment (anxolin) [inline src/scripts/deploy.js]: Not sure what will this version be, but If this is the version of the project, would it make sense to read from package.json or have it automatically:

https://docs.npmjs.com/cli/v6/using-npm/scripts#packagejson-vars

image

**Review by anxolin — COMMENTED** [review]
Cool! It would be great to have some DESCRIPTION in the PR and also something in the README

File: src/scripts/deploy.js (hunk 1 above)
Comment (tukantje) [reply src/scripts/deploy.js]: On it

Comment (tukantje) [reply src/scripts/deploy.js]: Good point - will do

File: package.json [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
   "license": "UNLICENSED",
+  "version": "0.0.1",
   "scripts": {
     "codegen": "graph codegen",
     "build": "graph build",
     "deploy": "node src/scripts/deploy.js",
-    "deploy:mainnet": "cross-env NETWORK=mainnet SUBGRAPH=gnosis/cow yarn deploy",
-    "deploy:rinkeby": "cross-env NETWORK=rinkeby SUBGRAPH=gnosis/cow-rinkeby yarn deploy",
-    "deploy:gc": "cross-env NETWORK=gc SUBGRAPH=gnosis/cow-gc yarn deploy"
+    "deploy:all": "run-s deploy:mainnet deploy:goerli deploy:gc",
```
Comment (alfetopito) [inline package.json:9]: What does `run-s` do? Where does it come from? (Where does it live? That, and much more, tonight at...)

File: subgraph.yaml.mustache [hunk 3]
```diff
@@ -27,12 +27,10 @@ dataSources:
           file: ./abis/GPV2Settlement.json
         - name: ERC20
           file: ./abis/ERC20.json
-        {{#usesUniV2}}
         - name: UniswapV2Factory
           file: ./abis/UniswapV2Factory.json
         - name: UniswapV2Pair
           file: ./abis/UniswapV2Pair.json
-        {{/usesUniV2}}
```
Comment (alfetopito) [inline subgraph.yaml.mustache:35]: So, this conditional is not needed anymore?

File: package.json (hunk 2 above)
Comment (tukantje) [reply package.json:9]: That is coming from `yarn-run-all` package. Short for run synchronously.

File: subgraph.yaml.mustache (hunk 3 above)
Comment (tukantje) [reply subgraph.yaml.mustache:35]: Yes - removed it on other places as well.

**Review by alfetopito — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -2,12 +2,11 @@
 
 Implements a subgraph for the [CoW Protocol](https://github.com/cowprotocol/contracts)
 
-*This is a work in progress*
+_This is a work in progress_
```
Comment (anxolin) [inline README.md:5]: Do we need to write this? maybe can delete

File: README.md [hunk 5]
```diff
@@ -2,12 +2,11 @@
...
 
-- [Subgraph on mainnet](https://thegraph.com/hosted-service/subgraph/cowprotocol/cow)
-- [Subgraph on rinkeby (will be deprecated)](https://thegraph.com/hosted-service/subgraph/cowprotocol/cow-rinkeby)
-- [Subgraph on goerli](https://thegraph.com/hosted-service/subgraph/cowprotocol/cow-goerli)
-- [Subgraph on gnosis chain network](https://thegraph.com/hosted-service/subgraph/cowprotocol/cow-gc)
+- [Subgraph on mainnet](https://thegraph.com/studio/subgraph/cow/)
+- [Subgraph on goerli](https://thegraph.com/studio/subgraph/cow-goerli/)
+- [Subgraph on gnosis chain network](https://thegraph.com/studio/subgraph/cow-gc/)
```
Comment (anxolin) [inline README.md:9]: maybe we could keep somewhere the "legacy hosted services endpoints"?

Comment (anxolin) [reply README.md:9]: If this is not adding any value, just ignore

File: src/scripts/deploy.js [hunk 6]
```diff
@@ -1,56 +1,93 @@
...
 
-const SUBGRAPH = process.env.SUBGRAPH
-assert(SUBGRAPH, 'SUBGRAPH env var is required')
+const SUBGRAPH = process.env.SUBGRAPH;
+assert(SUBGRAPH, "SUBGRAPH env var is required");
 
-console.log(`Prepare subgraph config for network ${NETWORK}`)
-console.log(`  Generate subgraph.yaml`)
+const AUTH_TOKEN = process.env.AUTH_TOKEN;
+assert(AUTH_TOKEN, "AUTH_TOKEN env var is required");
```
Comment (anxolin) [inline src/scripts/deploy.js:16]: same comment as before. Do we need AUTH_TOKEN? if so, shouldn't we have it specified in the README?

**Review by anxolin — COMMENTED** [review]
The deployed version don't show the USD price. 

There might be a misconfiguration

**Pull request merged**